cli/internal/context/context.go

```diff
@@ -419,3 +419,30 @@ func (c *Context) resolveDepGraph(wg *errgroup.Group, workspace *fs.PackageJSON,
 		})
 	}
 }
+
+// TopologicalGraphAncestors returns the ancestors of a set of nodes in the workspace graph.
+func (c *Context) TopologicalGraphAncestors(start []string) ([]string, error) {
+	vertices := make(dag.Set)
+	for _, v := range start {
+		vertices.Add(v)
+	}
+	s := make(dag.Set)
+	memoFunc := func(v dag.Vertex, d int) error {
+		s.Add(v)
+		return nil
+	}
+
+	if err := c.WorkspaceGraph.DepthFirstWalk(vertices, memoFunc); err != nil {
+		return nil, err
+	}
+
+	// Use for loop so we can coerce to string
+	// .List() returns a list of interface{} types, but
+	// we know they are strings.
+	targets := make([]string, 0, s.Len())
+	for _, dep := range s.List() {
+		targets = append(targets, dep.(string))
+	}
```

Comment:
Previously we always had the package being pruned as the [first element](https://github.com/vercel/turbo/blob/1b84680d1797f8a8ae11c2249ec6023ca8b60054/cli/internal/prune/prune.go#L146) and we have tests that reflect this behavior and are now breaking since the order of a set is non-deterministic.

The easiest solution is to use [`sort.Strings`](https://pkg.go.dev/sort#Strings) before returning to help out our tests (I'm pretty sure this also keeps the behavior our tests expect by chance).